src/presentation/transformers/UserTransformer.ts

```diff
@@ -19,6 +19,13 @@ export class UserTransformer {
     };
   }
 
+  public toRestrictedUserDTO(user: User): UserDTO {
```

Comment:
Should `RestrictedUserDTO` be a separate type?

Reply:
Yes, I ended up going that way, just pushed some updates.